Userland/Demos/Mandelbrot/Mandelbrot.cpp

```diff
@@ -203,15 +209,25 @@ void Mandelbrot::resize_event(GUI::ResizeEvent& event)
     m_set.resize(event.size());
 }
 
+void Mandelbrot::export_image(String const& export_path)
+{
+    auto png = Gfx::PNGWriter::encode(m_set.bitmap());
+    auto file = fopen(export_path.characters(), "wb");
+    fwrite(png.data(), 1, png.size(), file);
+    fclose(file);
+    GUI::MessageBox::show(window(), "Image was successfully exported.", "Mandelbrot", GUI::MessageBox::Type::Information);
+}
+
 int main(int argc, char** argv)
 {
     auto app = GUI::Application::construct(argc, argv);
 
-    if (pledge("stdio recvfd sendfd rpath", nullptr) < 0) {
+    if (pledge("stdio thread recvfd sendfd rpath wpath cpath", nullptr) < 0) {
         perror("pledge");
         return 1;
     }
 
+#if 0
```

Comment:
How about we just unveil the home directory for creation, and pop an error if they try to save somewhere else?

Reply:
Unveiling `/home`, `/etc/group` and `/etc/passwd` sort of works. However, when the user navigates outside of `/home` they end up with an empty file list. Also, this assumed the user's home directory is in `/home`. I think I'd rather wait for `FilePickerServer`. :)

Reply:
Sure, makes sense.